libs/deposits/src/lib/deposit-manager.tsx

```diff
@@ -0,0 +1,102 @@
+import type { DepositEvent_busEvents_event_Deposit } from './__generated__/DepositEvent';
+import { useEffect, useMemo, useState } from 'react';
+import { DepositForm } from './deposit-form';
+import { useBalanceOfERC20Token } from './use-balance-of-erc20-token';
+import { useDeposit } from './use-deposit';
+import sortBy from 'lodash/sortBy';
+import { useApprove } from './use-approve';
+import { useDepositLimits } from './use-deposit-limits';
+import { useAllowance } from './use-allowance';
+import { TransactionDialog } from '@vegaprotocol/ui-toolkit';
+import { useFaucet } from './use-faucet';
+import {
+  useTokenContract,
+  useBridgeContract,
+} from '@vegaprotocol/react-helpers';
+
+export interface Asset {
+  id: string;
+  symbol: string;
+  name: string;
+  decimals: number;
+  source: {
+    contractAddress: string;
+  };
+}
+
+interface DepositManagerProps {
+  requiredConfirmations: number;
+  bridgeAddress: string;
+  assets: Asset[];
+  initialAssetId?: string;
+}
+
+export const DepositManager = ({
+  requiredConfirmations,
+  bridgeAddress,
+  assets,
+  initialAssetId,
+}: DepositManagerProps) => {
+  const [assetId, setAssetId] = useState<string | undefined>(initialAssetId);
+
+  // Find the asset object from the select box
+  const asset = useMemo(() => {
+    const asset = assets?.find((a) => a.id === assetId);
+    return asset;
+  }, [assets, assetId]);
+
+  const tokenContract = useTokenContract(
+    asset?.source.contractAddress,
+    process.env['NX_VEGA_ENV'] !== 'MAINNET'
+  );
+  const bridgeContract = useBridgeContract();
+
+  // Get users balance of the erc20 token selected
+  const { balanceOf, refetch } = useBalanceOfERC20Token(tokenContract);
+
+  // Get temporary deposit limits
+  const limits = useDepositLimits(bridgeContract, asset);
+
+  // Get allowance (approved spending limit of brdige contract) for the selected asset
+  const allowance = useAllowance(tokenContract, bridgeAddress);
```

Comment:
chore (non-blocking): these two hooks are reading from the chain and the bottom 2 are writing to the chain. Could we have names that reflect that? Otherwise it looks like these are all the same and they aren't